core-java-modules/core-java-11-2/src/test/java/com/baeldung/features/JavaElevenFeaturesUnitTest.java

```diff
@@ -0,0 +1,61 @@
+package com.baeldung.features;
+
+import org.junit.jupiter.api.Test;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class JavaElevenFeaturesUnitTest {
+
+    @Test
+    void givenMultilineString_whenExtractingNonBlankStrippedLines_thenLinesAreReturned() {
+        String multilineString = "My name \n \n is \n Daniel.";
+        List<String> lines = multilineString.lines()
+                .filter(line -> !line.isBlank())
+                .map(String::strip)
+                .collect(Collectors.toList());
+        assertThat(lines).containsExactly("My name", "is", "Daniel.");
+    }
+
+    @Test
+    void givenTemporaryFile_whenReadingStringContent_thenContentIsReturned() throws IOException {
```

Comment:
Consider using https://www.baeldung.com/junit-5-temporary-directory here for a temp dir, rather then `Files.createTempFile` unless it's `Files.createTempFile` we're demoing!